src/main/java/com/minecolonies/coremod/entity/pathfinding/pathjobs/PathJobRandomPos.java

```diff
@@ -116,7 +114,8 @@ protected double computeHeuristic(@NotNull final BlockPos pos)
     @Override
     protected boolean isAtDestination(@NotNull final Node n)
     {
-        if ((minDistFromStart ==  1 || start.distSqr(n.pos) > minDistFromStart * minDistFromStart) && isWalkableSurface(world.getBlockState(n.pos.below()), n.pos.below()) == SurfaceType.WALKABLE
+        if (random.nextInt(10) == 0 && (start.distSqr(n.pos) > minDistFromStart * minDistFromStart)
+              && isWalkableSurface(world.getBlockState(n.pos.below()), n.pos.below()) == SurfaceType.WALKABLE
```

Comment:
random?

Reply:
ye to find a more spread out more random position